hera_cal/lst_stack/calibration.py

```diff
@@ -190,36 +208,50 @@ def _lstbin_amplitude_calibration(
             blpol = (ant1, ant2, pol)
 
             # Move to the next blpol if there is not a model for the data or the entire baseline is flagged
-            if np.all(stack.flags[:, apidx, :, polidx]):
+            if np.all(flags[:, apidx, :, polidx]):
                 continue
 
             # Get model, weights, and data for each baseline
             abscal_model[blpol] = model[apidx, :, polidx] * np.ones(
                 (len(stack.nights), 1)
             )
             data_here[blpol] = stack.data[:, apidx, :, polidx]
-            wgts_here[blpol] = stack.nsamples[:, apidx, :, polidx] * (
-                ~stack.flags[:, apidx, :, polidx]
+            wgts_here[blpol] = np.abs(stack.nsamples[:, apidx, :, polidx]) * (
```

Comment:
I suggest adding a comment on why you need to take the abs

Reply:
Added a comment here.